src/app/view/perspectives/smartdoc/bottom-panel/smartdoc-bottom-panel.component.ts

```diff
@@ -66,11 +66,13 @@ export class SmartDocBottomPanelComponent implements OnInit, OnDestroy {
     if (this.part.linkTypeId) {
       this.linkTypeSubscription = Observable.combineLatest(
         this.store.select(selectLinkTypeById(this.part.linkTypeId)),
-        this.store.select(selectCollectionsDictionary)
+        this.store.select(selectAllCollections).pipe(
+          map(collections => new Map(collections.map(collection => [collection.id, collection] as [string, CollectionModel])))
```

Comment:
Wouldn't it be better if this conversion was done in the subscribe method just before the data is needed? Because otherwise it will be done every time something is changed in collections store but it doesn't have to be processed later if the link is not present in the link store as we wait for it.

Reply:
Nice idea :)